cerberus/client.py

```diff
@@ -93,6 +195,82 @@ def get_sdb_id(self, sdb):
         # looking for.
         raise CerberusClientException("'%s' not found" % sdb)
 
+    def get_sdb_id_by_path(self, sdb_path):
```

Comment:
Why not just return the full object for the user to get the properties they want (e.g. `get_sdb_by_path`)?

Reply:
I was matching the style of `get_sdb_id`, and felt that it wasn't always desirable to track down an sdb by it's name if you're already dealing with paths.  
In testing I found myself using both `get_sdb_id` and `get_sdb_id_by_path` quite a bit.

But I could be thinking of things in a strange manner.

I considered writing a class that dealt with cerberus sdbs and secrets as objects, a bit like how boto works when you're constructing a route53 record.

Reply:
I don't think a classes for sdbs and secrets are necessary.

Also, there's nothing wrong with this method as is, but it allows more flexibility if the user can get the entire SDB object by the SDB path and then choose what fields they want to use

Reply:
Oh, I see what you're getting at.

`get_sdb_id` was already present in the code base so I was follow it's style, but I could add a functions like `get_sdb_by_id` and `get_sdb_by_path` that warp the `get_sdb_details` call.

Reply:
Right, the functionality of `get_sdb_details` is fine as is, but should be renamed to `get_sdb_by_id`.

Also `get_sdb_id_by_path` should just be `get_sdb_by_path`, even though there is already a `get_sdb_id`  ...by_name method

Reply:
I took a slightly different track to that.  But did change `get_sdb_details` to `get_sdb_by_id` and then added in some other functions to get the same object by supplying either a path or a name.

Reply:
I see. The reason I suggested changing this method is because now in the `get_sdb_by_path` method below you are calling GET twice for the same SDB.

If you want to keep the `get_sdb_id_by_path` method, then you should reverse the logic. `get_sdb_id_by_path` should call `get_sdb_by_path` instead of the other way around.

Reply:
I'll re-factor the functions a little bit.  Though to get the sdb details any way other than by id, it will always require two different calls. One to get the list and filter it by something, and the other to get the detail view of that sdb.

Though I could be missing something.

Reply:
You are right. That's my mistake. I forgot that listing SDBs only returns summaries instead of the full object. 

I will approve the PR. If you want, you can remove the `get_sdb_by_path` that you added. I was working under the wrong assumption when I suggested it.